verifier/verifier.go

```diff
@@ -518,6 +545,52 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r := revocation.New(client, logger)
+	authenticSigningTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	// TODO use authenticSigningTime from signerInfo
+	// https://github.com/notaryproject/notation-core-go/issues/38
+	revocationErr := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+
+	var importantErr bool
+	if revocationErr == nil {
+		importantErr = false
+	} else {
+		// If there is an error and it is designated as unimportant, log but pass the validation
+		switch t := revocationErr.(type) {
+		case ocsp.RevokedError:
+			importantErr = true
+		case ocsp.UnknownStatusError:
+			importantErr = true
+		case ocsp.NoOCSPServerError:
+			// Should this be an important error?
+			logger.Debugf("%v", t)
+			logger.Debug("1 or more certificates in the chain did not have OCSP configured. Ignoring these certificates")
+			importantErr = false
+		case ocsp.TimeoutError:
+			logger.Debugf("%v", t)
+			logger.Debug("Revocation unavailable. OCSP requests timed out")
+			importantErr = false
```

Comment:
is it right to fail open if revocation / ocsp can't be reached? this seems like an error that should fail the verification. users also have the option to fail open with trust policy level.